### DISCUSSION TITLE
DNS over TLS

### DISCUSSION THREAD

**Discussion by hros** [0xERR0R/blocky#824] [Q&A]
Is it possible to enable "private DNS" (DNS over TLS) with blocky
I am currently using blocky in my home network, and I have to disable Android's private-dns option to use blocky

**Comment by 0xERR0R** [comment]
Yes, you can use blocky as DoT resolver (aka private DNS) in android. You need valid SSL certificate for your domain (or a wildcard certificate). Some android versions have problems with newer LetsEncrypt certificates, you can try to use "ISRG Root X1" as preferred chain as a parameter for letsencrypt cert generation.

**Reply by hros** [reply]
Thanks
I have a valid certificate for my domain. The certificate was generated by cloudflare, I don't know what are the "ISRG Root X1" parameters of the certificate....
How do I configure blocky to use the certificate?

**Reply by 0xERR0R** [reply]
you need to define the "tlsPort" and also both certificate files: "certFile" and "keyFile" parameters. I would also recommend to check DoT with some external tool (like dog), because android has no debug information.
